### ISSUE TITLE
Unable to delete or modify private RPC nodes

### ISSUE THREAD

**Issue by benjaminion** [rotki/rotki#4769]
## Problem Definition

I have added two private Ethereum RPC nodes. When I try to delete or modify either of them I get a message "Request failed with status code 500" and the modifications fail.

I haven't tried messing with any of the public endpoints, so the issue may or may not also apply to them.

## Logs

This seems to be the relevant part of the log:

```
[30/08/2022 22:40:13 GMT Summer Time] DEBUG rotkehlchen.api.server Greenlet-6: start rotki api DELETE /api/1/blockchains/ETH/nodes view_args={}, query_string=b''
[30/08/2022 22:40:13 GMT Summer Time] CRITICAL rotkehlchen.api.server Greenlet-6: Unhandled exception when processing endpoint request exc_info=True, exception=[<class 'decimal.DivisionUndefined'>]
[30/08/2022 22:40:13 GMT Summer Time] DEBUG rotkehlchen.api.server Greenlet-6: end rotki api DELETE /api/1/blockchains/ETH/nodes view_args={}, query_string=b'', status_code=500, result={'result': None, 'message': "[<class 'decimal.DivisionUndefined'>]"}
[30/08/2022 22:40:13 GMT Summer Time] INFO geventwebsocket.handler 127.0.0.1 - - [2022-08-30 22:40:13] "DELETE /api/1/blockchains/ETH/nodes HTTP/1.1" 500 275 0.004886
```

### System Description

Operating system:  Windows
Rotki version: 1.25.2

**Comment by LefterisJP** [comment]
Hey @benjaminion Can you describe the steps you followed to get to this? How many nodes are in rotki, what weights and at what point the edit/delete got you there?

There is definitely a problem in the new logic we introduced there. But I hit something different :sweat_smile:

**Comment by benjaminion** [comment]
Cutting and pasting the node list from the app (can provide images if easier):

```
etherscan Etherescan API used by rotki Node weight 0%
mycrypto https://api.mycryptoapi.com/eth Node weight 0%
blockscout https://mainnet-nethermind.blockscout.com/ Node weight 0%
avado pool https://mainnet.eth.cloud.ava.do/ Node weight 0%
1inch https://web3.1inch.exchange Node weight 0%
myetherwallet https://names.mewapi.io/rpc/eth Node weight 0%
cloudflare https://cloudflare-eth.com/ Node weight 0%
my node https://mainnet.infura.io/v3/[redacted] Private nodes take priority over public nodes
MetalAlbert http://192.168.1.203:8545 Private nodes take priority over public nodes
```

The first 7 are automatically configured and I have not touched them. I did not change their weights myself, although they seem to have become set to zero at some point.

Quite a while ago I added "my node" which is an Infura endpoint.

Yesterday I successfully added "MetalAlbert".

I want to delete "my node", but trying to use the delete button gives the 500 error. Likewise anything I try to modify about the private nodes also gives 500.

**Comment by LefterisJP** [comment]
> The first 7 are automatically configured and I have not touched them. I did not change their weights myself, although they seem to have become set to zero at some point.

That's really weird.

Let us investigate, there seem to be multiple things a bit broken in 1.25.2 there.

 Scheduled it for our bugfixes branch but we most probably won't make 1.25.3 and it will be included in 1.26.0 directly.

**Comment by LefterisJP** [comment]
Fix is deployed with 1.25.3
